ros/src/computing/perception/detection/lidar_tracker/packages/lidar_imm_ukf_pda_track/nodes/imm_ukf_pda/ukf.cpp

```diff
@@ -1353,3 +1392,513 @@ void UKF::update(const bool use_sukf, const double detection_probability, const
+  if (model_ind == MotionModel::CV)
+  {
+    if (use_lane_direction)
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_direction_state_);
+      estimated_z << x_cv_(0), x_cv_(1), x_cv_(3);
+      epsilon = lidar_direction_cv_meas_ - estimated_z;
+      r = lidar_direction_r_cv_;
+    }
+    else
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_state_);
+      estimated_z << x_cv_(0), x_cv_(1);
+      epsilon = cv_meas_ - estimated_z;
+      r = r_cv_;
+    }
+    nis = nis_cv_;
+    x = x_cv_.col(0);
+    p = p_cv_;
+  }
+  else if (model_ind == MotionModel::CTRV)
+  {
+    if (use_lane_direction)
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_direction_state_);
+      estimated_z << x_ctrv_(0), x_ctrv_(1), x_ctrv_(3);
+      epsilon = lidar_direction_ctrv_meas_ - estimated_z;
+      r = lidar_direction_r_ctrv_;
+    }
+    else
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_state_);
+      estimated_z << x_ctrv_(0), x_ctrv_(1);
+      epsilon = ctrv_meas_ - estimated_z;
+      r = r_ctrv_;
+    }
+    nis = nis_ctrv_;
+    x = x_ctrv_.col(0);
+    p = p_ctrv_;
+  }
+  else
+  {
+    if (use_lane_direction)
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_direction_state_);
+      estimated_z << x_rm_(0), x_rm_(1), x_rm_(3);
+      epsilon = lidar_direction_rm_meas_ - estimated_z;
+      r = lidar_direction_r_rm_;
+    }
+    else
+    {
+      estimated_z = Eigen::VectorXd(num_lidar_state_);
+      estimated_z << x_rm_(0), x_rm_(1);
+      epsilon = rm_meas_ - estimated_z;
+      r = r_rm_;
+    }
+    nis = nis_rm_;
+    x = x_rm_.col(0);
+    p = p_rm_;
+  }
+
+  Eigen::MatrixXd x_sig = Eigen::MatrixXd(num_state_, 2 * num_state_ + 1);
+  Eigen::MatrixXd square_root_matrix_l = p.llt().matrixL();
+  x_sig.col(0) = x;
+  for (int i = 0; i < num_state_; i++)
+  {
+    Eigen::VectorXd sigma_point1 = x + sqrt(lambda_ + num_state_) * square_root_matrix_l.col(i);
+    Eigen::VectorXd sigma_point2 = x - sqrt(lambda_ + num_state_) * square_root_matrix_l.col(i);
+
+    sigma_point1(3) = normalizeAngle(sigma_point1(3));
+    sigma_point2(3) = normalizeAngle(sigma_point2(3));
+
+    x_sig.col(i + 1) = sigma_point1;
+    x_sig.col(i + 1 + num_state_) = sigma_point2;
+  }
+  Eigen::MatrixXd z_sig;
+  if (use_lane_direction)
+  {
+    z_sig = Eigen::MatrixXd(num_lidar_direction_state_, 2 * num_state_ + 1);
+  }
+  else
+  {
+    z_sig = Eigen::MatrixXd(num_lidar_state_, 2 * num_state_ + 1);
+  }
+  for (int i = 0; i < 2 * num_state_ + 1; i++)
+  {
+    double p_x = x_sig(0, i);
+    double p_y = x_sig(1, i);
+    z_sig(0, i) = p_x;
+    z_sig(1, i) = p_y;
+  }
+
+  Eigen::VectorXd z_pred;
+  if (use_lane_direction)
+  {
+    z_pred = Eigen::VectorXd(num_lidar_direction_state_);
+  }
+  else
+  {
+    z_pred = Eigen::VectorXd(num_lidar_state_);
+  }
+  z_pred.fill(0.0);
+  for (int i = 0; i < 2 * num_state_ + 1; i++)
+  {
+    z_pred = z_pred + weights_s_(i) * z_sig.col(i);
+  }
+
+  Eigen::MatrixXd s;
```

Comment:
@cirpue49 The CI error comes from this block, _s_ is being initialized by _Eigen::VectorXd_ shouldn't it be _Eigen::MatrixXd_?

Reply:
You are right.
Thank you for your comment!